### ISSUE TITLE
Loaded script do not redirect env variables on declarative

### ISSUE THREAD

**Issue by ismacasti** [jenkinsci/JenkinsPipelineUnit#346]
Hello,

I'm trying to test a declarative pipeline. The pipeline itself is as follows:
```
pipeline {
  [...]
  stages {
    stage("Prepare") {
      steps {
        load "whatever.groovy"
        load "animal_variables.groovy"
      }
    }
  }
}
```

And _variables.groovy_ is something like:
```
env.BIG_ANIMAL = "dog"
env.SMALL_ANIMAL = "cat"
env.MESSAGE = "${SMALL_ANIMAL} is eating"
```

This fails with:
```
No such property: SMALL_ANIMAL for class: animal_variables
groovy.lang.MissingPropertyException: No such property: SMALL_ANIMAL for class: animal_variables
```

Anywhere else, using the variables directly without using env works as in Jenkins, but it seems something does work there correctly.
I'm not familiar with the metaprogramming so help to get this working would be appreciated.

Thanks!

**Comment by nre-ableton** [comment]
The difference is that Jenkins will try to resolve `SMALL_ANIMAL` not only from `env`, but also from `params`. The safest way to make sure this works as expected would be:

```groovy
env.BIG_ANIMAL = "dog"
env.SMALL_ANIMAL = "cat"
env.MESSAGE = "${env.SMALL_ANIMAL} is eating"
```

**Comment by ismacasti** [comment]
But `SMALL_ANIMAL` is already on `env`, it was put there on the line immediately before. This feels like the mock resolver is not working.

**Comment by ismacasti** [comment]
I just tested this:
```
pipeline {
  [...]
    stages {
        stage("Prepare") {
            steps {
                script{
                    env.BIG_ANIMAL = "dog"
                    env.SMALL_ANIMAL = "cat"
                    env.MESSAGE = "${SMALL_ANIMAL} is eating"
                    echo "${MESSAGE}"
                }
                load "whatever.groovy"
                load "animal_variables.groovy"
[...]
```
And this works fine, it even prints `cat is eating` on the console. I think the methodInterceptor doesn't get injected on the loaded script.

**Comment by nre-ableton** [comment]
Yes, you're right, it seems to be an issue with `loadInterceptor`.

**Comment by ismacasti** [comment]
I'm not familiar with the code, any tips to start looking into fixing this?

**Comment by nre-ableton** [comment]
I'm also not exactly sure, to be honest. I haven't dabbled in this area of the codebase. @stchar -- might you know?

**Comment by ismacasti** [comment]
I found a way, maybe it's a bad hack but seems to work.
Just implement into the loaded script object the metaclass.propertyMissing (and static) and in the closure redirect to the env variable in the binding.

I got the idea here:
https://e.printstacktrace.blog/groovy-static-propertymissing-and-methodmissing-methods-limitations-and-possible-issues/

**Comment by nre-ableton** [comment]
@ismacasti Great! Do you mind if I close this issue then?
